lib/syskit/models/composition.rb

```diff
@@ -352,6 +357,42 @@ def add_main(models, **options)
                 @main_task = add(models, **options)
             end
 
+            class PortWriter
+                attr_reader :child_name
+                attr_reader :port_name
+
+                def initialize(port)
+                    @child_name = port.component_model.child_name
+                    @port_name = port.name
+                end
+
+                def bind(cmp)
+                    @writer = cmp.child_from_role(child_name)
+                                 .find_input_port(port_name)
+                                 .writer
+                end
+
+                def write(sample)
+                    @writer.write(sample)
+                end
+            end
+
+            def port_writer_by_name(name)
+                @port_writers[name.to_sym]
```

Comment:
`@port_writers` will be nil here if called before `#port_writers` has ever been called. Just use the method version which will auto-initialize the 

Generally speaking, I'm not sure what this attribute brings ? The port could be resolved and memoized straight in the generated method.

Reply:
> will be nil here if called before #port_writers has ever been called

Good point.. :+1: 

> I'm not sure what this attribute brings 

Seemed consistent with how everything else is stored. I can change to a plain instance variable though, no preference.

> The port could be resolved and memoized straight in the generated method.

In the generated method? And where would I keep the arguments passed in the model's `port_writer` helper (the port object, so far)